flake8_type_checking/checker.py

```diff
@@ -932,8 +977,12 @@ def __init__(
         #: List of all names and ids, except type declarations
         self.uses: dict[str, list[tuple[ast.AST, Scope]]] = defaultdict(list)
 
+        #: Contains a set of all names to be treated like soft-uses.
+        # i.e. we don't know if it will be used at runtime or not
+        self.soft_uses: set[str] = set()
```

Comment:
For clarification, could you elaborate a little bit on when we do flag soft uses as errors?

Reply:
We don't ever flag them, their purpose is to avoid flagging entirely, since we can't know whether they should be available at runtime.

E.g. for SQAlchemy you need most things inside `Mapped[]` to be available at runtime. But for references to other models you're allowed to have the imports inside a `if TYPE_CHECKING` block, because cyclic dependencies are quite common with models and SQLAlchemy already needs to keep track of all the models anyways, so those forward references can be resolved, even if there is no runtime import. But we have no way of knowing which uses are models and which ones are just random other types.

So a soft use is for cases where we need to refuse the temptation to guess and trust the code is correct as it is.

Reply:
But I just noticed I did the wrong thing with `Annotated`, I had intended to treat the first argument as a soft use, since it is quite likely that `Annotated` types will be used at runtime, and the rest of the arguments as regular uses for runtime annotations, and soft uses when `from __future__ import annotations` is present. Treating the rest of the arguments as soft uses only makes sense when the annotation is deferred.